specs/Alert_Notifications.adoc

```diff
@@ -0,0 +1,186 @@
+
+== Use Cases
+
+Goal: Notify user in case of any alert event.
+
+Actors:
+
+1) User
+
+Steps:
+
+* Node Agent stores alert events in etcd.
+
+* The alert events will be fetched by the Notifier.
+
+* The alert event is to be sent to respective user via configured
+   channels .
+
+== Proposed change
+
+
+Present Alert system(https://github.com/Tendrl/alerting) will be renamed as “Notifier”
+
+The main functionality of Notifier would be to:
+
+
+Monitor etcd only for alert specific events.
+
+Sending alert notifications to configurable receiving end-points
+
+Presently the alerting module monitors all events and looks for events having
+priority as “notice” which signifies they are alerts, which will be changed to
+monitoring just the alert specific events populated by node-agent.
+
+The Notifier can receive the alert events in 2 ways:
+
+* The alert specific events are stored in etcd by the node-agent. The notifier could
+   collect the events from etcd and then using the configs in etcd it could 
+   notify the respective user.
+
+ Advantage:
+ 
+     ** Simple architecture and easy to implement.
+
+  Drawback:
+ 
+      ** Pooling etcd for new alerts is not efficient.
+ 
+* The node-agent can directly pass the alert events to the notifier via an endpoint
+   exposed by the notifier.
+
+    Advantage:
+ 
+     ** Avoids pooling etcd for new alerts.
+ 
+    Drawbacks:
+
+     **  Need to create and maintain an event receiver in existing alerting system
+
+The Notifier will notify the user of the events via configured notification channels
+like SMTP.
+
+=== Alternatives:
+
+None
+
+
+=== Data model impact:
+
+ No changes in existing structure.
+
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+None
+
+==== Notifications/Monitoring impact:
+
+None
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+Node-agent must be modified to insert the events received by Monitoring
+Integration in etcd under alerting/alerts/ instead of messages/events/.
+
+Monitoring integration adds scope ( global, cluster specific or node specific)
+of alert in the message metadata, this scope should be used by node-agent
+to place the event in the proper path in etcd.
+ 
+
+==== Sds integration impact:
+
+None
+
+==== Tendrl/performance-monitoring impact:
+
+None
+
+==== Tendrl/alerting impact:
+
+Tendrl alerting system will be renamed as Notifier.
+Its main functionality would be to fetch the alert events and
+notify the respective user.
+
+=== Security impact:
+ 
+ None
+ 
+=== Other end user impact:
+
+Users will be able to set alerts in Grafana itself and get notified for the same.
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+Modify node-agent to insert events into different path in etcd.
+
+Trim down present alerting system to provide only notification
+            functionality.
+
+== Implementation:
+
+WIP
+
+=== Assignee(s):
+
+
+Gowtham S
+Rishubh jain
```

Comment:
I dont see a mention about clearing alerts (say like we raised and alert saying CPU uses at the moment is 90%. But later when it goes down there should be a clearing alert for this existing one. This alert should vanish/or should not keep listing in tendrl UI).

If clearing is not provided user would manually need to correlate between threshold cross and then within limits cases.